### ISSUE TITLE
Trying to use string enum for key type of Immutable.Map

### ISSUE THREAD

**Issue by flarnie** [facebook/immutable-js#1371]
Hi - I'm updating Draft.js to use Immutable.js v3.8.1, and we happen to have a type defined as follows:

```
export type DraftBlockRenderMap = Map<DraftBlockType, DraftBlockRenderConfig>;
```
https://github.com/facebook/draft-js/blob/master/src/model/immutable/DraftBlockRenderMap.js#L19

In this definition `DraftBlockType` is a string enum, and we get an error that it is incompatible with the `string` type required for a key in an Immutable.js map:
```
475:   static <V>(obj?: {[key: string]: V}): Map<string, V>;

                                                 ^^^^^^ string. This type is incompatible with

 24: export type DraftBlockRenderMap = Map<DraftBlockType, DraftBlockRenderConfig>;

                                           ^^^^^^^^^^^^^^ string enum. See: src/model/immutable/DraftBlockRenderMap.js:24

```
https://travis-ci.org/facebook/draft-js/jobs/285870587

This makes sense, because in Flow you can't reconcile a string enum with a string type requirement. [This issue](https://github.com/facebook/flow/issues/2436) explains a bit.

I think this may be a common use case, and I'm wondering what advice folks have about how to fix it. It would be nice to have the Map type definition allow for string enums but I'm not sure how to do that.

This is, I think, the last error blocking [our upgrade in Draft.js](https://github.com/facebook/draft-js/pull/1289).

**Comment by leebyron** [comment]
Thanks for reporting this!

I think this may be one of our last holdovers from older flow versions, which always assumed object keys were strings - though now they can be more specific types.

I'll work on a fix and get a new RC out to unblock draft.js!

**Comment by leebyron** [comment]
Oh I just noticed you mentioned it's bumping up to `v3.8.1`. I'm currently only working on the flow definitions for v4 and not patching changes to older versions. I'll get this fixed in v4, but you should probably use any-casting or FlowFixMe for this particular case to unblock Draft

**Comment by flarnie** [comment]
Thanks! Sounds good, and I look forward to v4. Will be trying out the RC when I have time. :)
